### PR TITLE
Fix the TimeProvider test

### PR DESCRIPTION
**Pull request by tarekgh**

### CODE REVIEW

**Comment by tarekgh** [comment]
https://github.com/dotnet/runtime/pull/85249#issuecomment-1520333273

**Comment by tarekgh** [comment]
@stephentoub could you please have a quick look? Thanks!

File: src/libraries/Common/tests/System/TimeProviderTests.cs [hunk 1]
```diff
@@ -140,7 +140,7 @@ public void TestProviderTimer(TimeProvider provider, int MaxMilliseconds)
                             state,
                             TimeSpan.FromMilliseconds(state.Period), TimeSpan.FromMilliseconds(state.Period));
 
-            state.TokenSource.Token.WaitHandle.WaitOne(60000);
+            Assert.True(state.TokenSource.Token.WaitHandle.WaitOne(Timeout.InfiniteTimeSpan));
```
Comment (stephentoub) [inline src/libraries/Common/tests/System/TimeProviderTests.cs:143]: The assert here is meaningless if an infinite timeout is used, as WaitOne only returns false if it times out.

Comment (tarekgh) [reply src/libraries/Common/tests/System/TimeProviderTests.cs]: Ok, I removed the assert and kept the timeout to be infinite.

**Comment by lewing** [comment]
please don't try to fix the cla check here, I'm having it looked at

**Comment by lewing** [comment]
investigation is finished

**Review by stephentoub — APPROVED** [review]

**Pull request merged**